enterprise/frontend/src/embedding-sdk/components/private/util/MultiStepPopover/MultiStepPopover.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+  Children,
+  type PropsWithChildren,
+  type ReactElement,
+  type ReactNode,
+  isValidElement,
+} from "react";
+
+import { Paper, Popover } from "metabase/ui";
+
+const Step = ({ children }: StepProps) => {
+  return <Paper className="p-4">{children}</Paper>;
+};
+
+type StepProps = {
+  value: string | number;
+  children: ReactNode;
+};
+
+const isStepElement = (
+  element: ReactElement,
+): element is ReactElement<StepProps> => {
+  return element.type === Step;
+};
+
+type MultiStepPopoverTargetProps = {
+  children: ReactNode;
+};
+
+const Target = ({ children }: MultiStepPopoverTargetProps) => {
+  return <>{children}</>;
+};
+
+const MultiStepPopover = ({
+  currentStep,
+  children,
+}: PropsWithChildren<{
+  currentStep: string | number;
+}>) => {
+  const getCurrentStepContent = () => {
+    return (
+      Children.toArray(children)
+        .filter((child): child is ReactElement => isValidElement(child))
+        .find(
+          child => isStepElement(child) && child.props.value === currentStep,
+        )?.props.children || null
```

Comment:
I was thinking if there is [an alternative to Children.toArray](https://react.dev/reference/react/Children#alternatives) for this, but given the API this seems to be the best way to go.